src/main/java/appeng/parts/reporting/ConversionMonitorPart.java

```diff
@@ -18,25 +18,37 @@
 
 package appeng.parts.reporting;
 
+import org.jetbrains.annotations.Nullable;
+
 import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.world.InteractionHand;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.phys.Vec3;
 
+import appeng.api.networking.IStackWatcher;
+import appeng.api.networking.crafting.ICraftingWatcherNode;
 import appeng.api.parts.IPartItem;
 import appeng.api.parts.IPartModel;
 import appeng.api.stacks.AEItemKey;
+import appeng.api.stacks.AEKey;
+import appeng.api.storage.ITerminalHost;
+import appeng.api.storage.MEStorage;
 import appeng.api.storage.StorageHelper;
+import appeng.api.util.IConfigManager;
 import appeng.core.AppEng;
 import appeng.items.parts.PartModels;
 import appeng.me.helpers.PlayerSource;
+import appeng.menu.ISubMenu;
+import appeng.menu.locator.MenuLocators;
+import appeng.menu.me.crafting.CraftAmountMenu;
 import appeng.parts.PartModel;
 import appeng.util.InteractionUtil;
 import appeng.util.Platform;
 import appeng.util.inv.PlayerInternalInventory;
 
-public class ConversionMonitorPart extends AbstractMonitorPart {
+public class ConversionMonitorPart extends AbstractMonitorPart implements ITerminalHost {
```

Comment:
Why `ITerminalHost`? Would it not be enough to make `CraftAmountMenu` support any `ISubMenuHost` instead?